File: python/problem1411.py
class Solution:
    def numOfWays(self, n: int) -> int:
        ABC, ABA = 6, 6
        mod = 1000000007
        for _ in range(n - 1):
            ABC, ABA = 2 * ABC + 2 * ABA, 2 * ABC + 3 * ABA
        return (ABC + ABA) % mod

File: Java/problem1411.java
class Solution {
    
    public int numOfWays(int n) {
        int ABC = 6, ABA = 6;
        int mod = 1_000_000_007;
        for (int i = 1; i < n; ++i) {
            int temp = ABC;
            ABC = (int)((ABC * 2L + ABA * 2L) % mod);
            ABA = (int)((temp * 2L + ABA * 3L) % mod);
        }
        return (int)(ABC + ABA) % mod;
    }
}
